### ISSUE TITLE
Automatic refactoring: changing variable names

### ISSUE THREAD

**Issue by CameronBieganek** [fonsp/Pluto.jl#2209]
It would be awesome if Pluto automatically (reactively) updated variable names when you change the name of a variable where it is first defined. For example, if I had these two cells,

### Cell 1
```julia
a = 1
```

### Cell 2
```julia
b = a + 42
```

and I changed the name of `a` to `foo` in Cell 1, it would be great if Pluto automatically updated Cell 2 to read `b = foo + 42`.

**Comment by fonsp** [comment]
@Pangoraw or @dralletje have we worked on this before? How feasible is this? What about macros like `@syms x y z`?

**Comment by Pangoraw** [comment]
> How feasible is this?

If we can add back something like #1925 with a smaller performance cost it should be quite easy to implement (not for macro defined symbols however).

**Comment by dralletje** [comment]
> > How feasible is this?
> 
> If we can add back something like #1925 with a smaller performance cost it should be quite easy to implement (not for macro defined symbols however).

Yeah, this would be best I think. We could add some special cases for `@gensym` and friends, but there is no way to backtrack what symbol in the code was used to create a variable in a macro result (sadly), so no automagic here :(

Would be cool if we had some flow for applying these kind of tasks to your notebook, where Pluto would present you with cases where it is not sure what to do (say, we know a variable is used in a cell, but we can't figure out where so we show it to you like "Hey you need to update something here!"), but that would be a whole project 😅

**Comment by Astrusia** [comment]
would be nice to see this revisited. since the interface is in a browser, which does not have a find and replace, only find (unlike pretty much every other editor)
